### PR TITLE
Using RskAddress in Genesis premine map

### PR DESCRIPTION
**Pull request by ajlopezrsk**
Instead of ByteArrayWrapper

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [BlockChainLoader.java#L27](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java#L27): Unused import - org.ethereum.db.ByteArrayWrapper. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [GenesisLoader.java#L31](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L31): Unused import - org.ethereum.db.ByteArrayWrapper. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [Genesis.java#L62](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/Genesis.java#L62): Constructor has 16 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L45](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L45): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L75](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L75): Replace this use of System.out or System.err by a logger. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L77](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L77): Remove this call to "exit" or ensure it is really required. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1147)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L104](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L104): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L121](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L121): Iterate over the "entrySet" instead of the "keySet". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2864)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L122](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L122): org.ethereum.core.genesis.GenesisLoader.generateRootHash(Map) makes inefficient use of keySet iterator instead of entrySet iterator [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)
1. ![MINOR][MINOR] [BlockChainLoader.java#L27](https://github.com/rsksmart/rskj/blob/64556dd3e6798ff7d80cb60db9adff14b95a32f1/rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java#L27): Remove this unused import 'org.ethereum.db.ByteArrayWrapper'. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java [hunk 1]
```diff
@@ -79,41 +79,47 @@ public static Genesis loadGenesis(RskSystemProperties config, BigInteger initial
...
+
                     if (contractDetails.getCode() != null) {
                         acctState.setCodeHash(sha3(contractDetails.getCode()));
                     }
+
                     acctState.setStateRoot(contractDetails.getStorageHash());
                 }
-                premine.put(wrap(Hex.decode(accountEntry.getKey())), new InitialAddressState(acctState, contractDetails));
+
+                premine.put(new RskAddress(Hex.decode(accountEntry.getKey())), new InitialAddressState(acctState, contractDetails));
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java]: Please use the other constructor: `new RskAddress(accountEntry.getKey())`

File: rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java [hunk 2]
```diff
@@ -69,19 +69,20 @@ public void loadBlockchain() {
 
             BigInteger initialNonce = config.getBlockchainConfig().getCommonConstants().getInitialNonce();
             Genesis genesis = GenesisLoader.loadGenesis(config, config.genesisInfo(), initialNonce, true);
-            for (ByteArrayWrapper address : genesis.getPremine().keySet()) {
-                RskAddress addr = new RskAddress(address.getData());
-                repository.createAccount(addr);
+            for (RskAddress address : genesis.getPremine().keySet()) {
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java]: I'd keep `addr` name for consistency.

Comment (aeidelman) [reply rskj-core/src/main/java/org/ethereum/core/genesis/BlockChainLoader.java]: @ajlopezrsk please use 'addr', it's the name we're using everywhere else

File: rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java (hunk 1 above)
Comment (ajlopezrsk) [reply rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java]: done!

File: rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java [hunk 3]
```diff
@@ -79,41 +79,47 @@ public static Genesis loadGenesis(RskSystemProperties config, BigInteger initial
...
     }
 
-    private static byte[] generateRootHash(Map<ByteArrayWrapper, InitialAddressState> premine){
-
+    private static byte[] generateRootHash(Map<RskAddress, InitialAddressState> premine){
         Trie state = new TrieImpl(null, true);
 
-        for (ByteArrayWrapper key : premine.keySet()) {
-            state = state.put(key.getData(), premine.get(key).getAccountState().getEncoded());
+        for (RskAddress address : premine.keySet()) {
```
Comment (aeidelman) [inline rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java]: Please use 'addr' as name here

File: rskj-core/src/test/java/co/rsk/core/bc/BlockChainImplTest.java [hunk 4]
```diff
@@ -906,9 +905,9 @@ public static Block getGenesisBlock(BlockChainImpl blockChain) {
 
         Genesis genesis = GenesisLoader.loadGenesis(config, "rsk-unittests.json", BigInteger.ZERO, true);
 
-        for (ByteArrayWrapper key : genesis.getPremine().keySet()) {
-            repository.createAccount(new RskAddress(key.getData()));
-            repository.addBalance(new RskAddress(key.getData()), genesis.getPremine().get(key).getAccountState().getBalance());
+        for (RskAddress address : genesis.getPremine().keySet()) {
```
Comment (aeidelman) [inline rskj-core/src/test/java/co/rsk/core/bc/BlockChainImplTest.java]: Please use 'addr' as name here

**Comment by rskops** [comment]
SonarQube analysis reported 14 issues
* ![CRITICAL][CRITICAL] 2 critical
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 3 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [GenesisLoader.java#L31](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L31): Unused import - org.ethereum.db.ByteArrayWrapper. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [GenesisLoader.java#L36](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L36): Unused import - org.spongycastle.util.encoders.Hex. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [Genesis.java#L62](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/Genesis.java#L62): Constructor has 16 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L45](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L45): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L75](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L75): Replace this use of System.out or System.err by a logger. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L77](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L77): Remove this call to "exit" or ensure it is really required. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1147)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L104](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L104): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L121](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L121): Iterate over the "entrySet" instead of the "keySet". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2864)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L122](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L122): org.ethereum.core.genesis.GenesisLoader.generateRootHash(Map) makes inefficient use of keySet iterator instead of entrySet iterator [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)
1. ![MINOR][MINOR] [GenesisLoader.java#L31](https://github.com/rsksmart/rskj/blob/cce8f0198142b46aac4ff565c3981341db0ac650/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L31): Remove this unused import 'org.ethereum.db.ByteArrayWrapper'. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 25 issues
* ![CRITICAL][CRITICAL] 3 critical
* ![MAJOR][MAJOR] 13 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 7 info

#### Top 10 issues

1. ![CRITICAL][CRITICAL] [Wallet.java#L217](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/co/rsk/core/Wallet.java#L217): Object deserialization is used in co.rsk.core.Wallet.decryptAES(byte[], byte[]) [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findsecbugs%3AOBJECT_DESERIALIZATION)
1. ![CRITICAL][CRITICAL] [GenesisLoader.java#L31](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L31): Unused import - org.ethereum.db.ByteArrayWrapper. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![CRITICAL][CRITICAL] [GenesisLoader.java#L36](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L36): Unused import - org.spongycastle.util.encoders.Hex. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck)
1. ![MAJOR][MAJOR] [Genesis.java#L62](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/Genesis.java#L62): Constructor has 16 parameters, which is greater than 7 authorized. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00107)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L45](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L45): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L75](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L75): Replace this use of System.out or System.err by a logger. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS106)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L77](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L77): Remove this call to "exit" or ensure it is really required. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1147)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L104](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L104): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L121](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L121): Iterate over the "entrySet" instead of the "keySet". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS2864)
1. ![MAJOR][MAJOR] [GenesisLoader.java#L122](https://github.com/rsksmart/rskj/blob/1a67f3983aa9af0c5c443b4cff18c759ee6a8383/rskj-core/src/main/java/org/ethereum/core/genesis/GenesisLoader.java#L122): org.ethereum.core.genesis.GenesisLoader.generateRootHash(Map) makes inefficient use of keySet iterator instead of entrySet iterator [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=findbugs%3AWMI_WRONG_MAP_ITERATOR)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**